ansible/roles/svc_fluentd/tasks/setup_fluentd.yml

```diff
@@ -0,0 +1,66 @@
+---
+# code: language=ansible
+# [Doc] List of all modules used
+# File https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html
+# Template https://docs.ansible.com/ansible/latest/collections/ansible/builtin/template_module.html
+# Docker_image_pull https://docs.ansible.com/ansible/latest/collections/community/docker/docker_image_pull_module.html
+# Docker_volume https://docs.ansible.com/ansible/latest/collections/community/docker/docker_volume_module.html
+# Docker_container https://docs.ansible.com/ansible/latest/collections/community/docker/docker_container_module.html
+
+- name: Pull fluentd image
+  community.docker.docker_image_pull:
+    name: "{{ fluentd_docker_image_name }}"
+    tag: "{{ fluentd_docker_image_version }}"
+    pull: always
+    timeout: 10
+    tls: true
+    tls_hostname: "{{ docker_registry_hostname }}"
+  become: true
+
+- name: Create directory for fluentd
+  ansible.builtin.file:
+    name: "{{ fluentd_data_dir }}"
+    state: directory
+    mode: "700"
+    recurse: true
+  become: true
+
+- name: Deploy fluentd Configuration to Host
+  ansible.builtin.template:
+    src: templates/fluentd.yml.j2
+    dest: "{{ fluentd_data_dir }}/fluentd.yml"
+    mode: "600"
+  become: true
+
+- name: Create docker_volume for fluentd data
+  community.docker.docker_volume:
+    name: "{{ fluentd_docker_volume_name }}"
+    driver: local
+    driver_options:
+      type: "none"
+      o: "bind"
+      device: "{{ fluentd_data_dir }}"
+    state: present
+  become: true
```

Comment:
You are not using the docker_volume here ( your are directly just mounting the config files).
You must remove the volume creation and linked configs.